lib/ansible/modules/cloud/kubevirt/kubevirt_vm.py

```diff
@@ -414,7 +414,7 @@ def execute_module(self):
         # Changes in VM's spec or any changes to VMIs warrant a full CRUD, the latter because
         # VMIs don't really have states to manage; they're either present or don't exist
         # Also check_mode always warrants a CRUD, as that'll produce a sane result
-        if vm_spec_change or ephemeral or k8s_state == 'absent' or self.check_mode:
+        if vm_spec_change or (ephemeral and vm_spec_change) or k8s_state == 'absent' or self.check_mode:
```

Comment:
Now that my brain is back, I'm pretty sure your code should be this:
`if vm_spec_change or k8s_state == 'absent' or self.check_mode:`
Unless I'm mistaken, please correct the patch and I'll +1 it. You're right that with the current logic, running crud just because the VM is ephemeral is pointless.

Reply:
It shouldn't because then ephemearl VMs won't work.